### ISSUE TITLE
Restoring cache fails with pyproject.toml

### ISSUE THREAD

**Issue by garymm** [prefix-dev/setup-pixi#93]
Say my repo is named `foo`, and inside the repo I have another directory named `foo`.
At the root of the repo I have a `pyproject.toml` file that contains:

```toml
[build-system]
requires = ["setuptools"]
build-backend = "setuptools.build_meta"

[project]
name = "foo"
requires-python = ">=3.10"
dynamic = ["version"]

[tool.pixi.project]
name = "foo"
channels = ["conda-forge"]

[tool.pixi.pypi-dependencies]
foo = { path = "./", editable = true }
```

When I use the setup-pixi action version 0.5.1 it fails:
```
Restoring pixi cache
Error: Failed to generate cache key: Error: ENOENT: no such file or directory, open '/home/runner/work/foo/foo/pyproject.lock'
```

GitHub job:
```yaml
jobs:
  pytest:
    runs-on: ubuntu-22.04
    steps:
      - uses: actions/checkout@v4
      - uses: prefix-dev/setup-pixi@v0.5.1
        with:
          manifest-path: pyproject.toml
          pixi-version: v0.18.0
          cache: true
          cache-write: ${{ github.event_name == 'push' && github.ref_name == 'main' }}
      - name: pytest
        run: pixi run pytest --capture=no --verbose
```

**Comment by ruben-arts** [comment]
The action doesn't work by default with the pyproject.toml yet. You have to give it a "manifest_path" for now.

**Comment by garymm** [comment]
I did.

**Comment by ruben-arts** [comment]
Could you share your GitHub action?

**Comment by garymm** [comment]
@ruben-arts added to the description

**Comment by ruben-arts** [comment]
Thanks, it looks like everything should work as expected. Maybe @pavelzw has an idea on top of his mind? Otherwise I can check when I'm back at my pc.

**Comment by pavelzw** [comment]
yeah, i think whats happening is that i try to get the path of pixi.lock by removing the .toml part and replacing it with .lock 😅 this obviously is a wrong assumption with the new pixi version 😅 i won't be able to fix this issue this week but i can take a look next week (and actually do proper pyproject.toml manifest support while i'm at it).
until this fix landed, you could just use `cache: false`

**Comment by pavelzw** [comment]
https://github.com/prefix-dev/setup-pixi/blob/92e1dd3a46d4cb599eccd780b6a485a7deb9dab0/src/options.ts#L170 is the relevant line

**Comment by pavelzw** [comment]
we probably didn't catch it in #87 because there cache: false was implicitly chosen because `pyproject.lock` was not found

**Comment by pavelzw** [comment]
Fixed by #95
